programs/bpf/rust/instruction_introspection/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+//! @brief Example Rust-based BPF program that exercises instruction introspection
+
+extern crate solana_sdk;
+use solana_sdk::{
+    account_info::next_account_info, account_info::AccountInfo, entrypoint,
+    entrypoint::ProgramResult, info, instruction::Instruction, program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+entrypoint!(process_instruction);
+fn process_instruction(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    if instruction_data.is_empty() {
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    let secp_instruction_index = instruction_data[0];
```

Comment:
Probably want the index to be part of the sysvar rather than having it come in as part of the instruction data